File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    answer = 0
    for i in range(left,right+1):
        div = 0
        for j in range(1,i+1):
            if i%j ==0:
                div +=1
        if div %2 ==0:
            answer +=i
        else:
            answer -=i
    return answer
# 모든 제곱수의 약수의 갯수는 홀수, 나머지 수는 짝수이다...

File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.js
function solution(left, right) {
    let answer = 0;
   
    for (let i= left;i<=right;i++){
        let div = 0;

        for (let j = 1;j*j<=i;j++)
            if (i%j ==0){
                div+=2;
                
                if (j*j==i){
                    div -=1
                    
                }}
        if (div %2 ==0){
            answer +=i
        }
        else{
            answer -=i
        }
        
        
        }
    return answer;
}